Bottom: aadbdb59a030a4ac9d359d607858445f547866b2
Top:    8c99bb48507d06cdf883f91641ca9d27a11700d0
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-05-21 19:15:10 -0700

Refresh of pre-qpsize

---

diff --git a/src/preload.c b/src/preload.c
index 12f2c02..1214bb5 100644
--- a/src/preload.c
+++ b/src/preload.c
@@ -308,17 +308,11 @@ err:
  */
 void set_rsocket_options(int rsocket)
 {
-	socklen_t len;
+	if (sq_size)
+		rsetsockopt(rsocket, SOL_RDMA, RDMA_SQSIZE, &sq_size, sizeof sq_size);
 
-	if (sq_size) {
-		len = sizeof(sq_size);
-		rsetsockopt(rsocket, SOL_RDMA, RDMA_SQSIZE, &sq_size, &len);
-	}
-
-	if (rq_size) {
-		len = sizeof(rq_size);
-		rsetsockopt(rsocket, SOL_RDMA, RDMA_RQSIZE, &rq_size, &len);
-	}
+	if (rq_size)
+		rsetsockopt(rsocket, SOL_RDMA, RDMA_RQSIZE, &rq_size, sizeof rq_size);
 }
 
 int socket(int domain, int type, int protocol)
